### PR TITLE
feat(cli): add close publisher caps

### PR DESCRIPTION
**Pull request by keyvankhademi**

### CODE REVIEW

File: staking/cli/src/instructions.rs [hunk 1]
```diff
@@ -870,6 +878,191 @@ pub fn update_y(rpc_client: &RpcClient, signer: &dyn Signer, y: u64) {
...
+    data.extend_from_slice(&[1]);
+    data.extend_from_slice(&signer.pubkey().to_bytes());
+
+    rpc_client
+        .get_program_accounts_with_config(
+            &publisher_caps::ID,
+            RpcProgramAccountsConfig {
+                filters:        Some(vec![RpcFilterType::Memcmp(Memcmp::new(
+                    0,
+                    MemcmpEncodedBytes::Bytes(PublisherCaps::DISCRIMINATOR.to_vec()),
```
Comment (guibescos) [inline staking/cli/src/instructions.rs:892]: you should filter on `data` here

File: staking/cli/src/main.rs [hunk 2]
```diff
@@ -97,5 +99,14 @@ fn main() {
         Action::SaveStakeAccountsSnapshot {} => {
             save_stake_accounts_snapshot(&rpc_client);
         }
+        Action::ClaimRewards {
+            min_staked,
+            min_reward,
```
Comment (guibescos) [inline staking/cli/src/main.rs:104]: IMO you should just pick one of the two parameters and drop the other one.

File: staking/cli/src/instructions.rs [hunk 3]
```diff
@@ -870,6 +878,191 @@ pub fn update_y(rpc_client: &RpcClient, signer: &dyn Signer, y: u64) {
...
+            }
+
+            println!(
+                "Advance delegation record for pubkey: {:?} publisher: {:?} with reward: {:?}",
+                positions.to_string(),
+                publisher,
+                reward_amount,
+            );
+
+            process_transaction(rpc_client, &[ix], &[signer]).unwrap();
```
Comment (guibescos) [inline staking/cli/src/instructions.rs:1005]: should we add `merge_delegation_positions` here

File: staking/cli/src/main.rs (hunk 2 above)
Comment (guibescos) [inline staking/cli/src/main.rs:104]: I don't like having two parameters here. I feel like we should be disciplined and:
- either claim all rewards for accounts with min_staked
- or claim all rewards s.t. reward > min_reward

File: staking/cli/src/cli.rs [hunk 4]
```diff
@@ -110,6 +110,13 @@ pub enum Action {
         publisher_caps: Pubkey,
     },
     SaveStakeAccountsSnapshot {},
+    ClaimRewards {
+        #[clap(long, help = "Minimum staked tokens")]
+        min_staked: u64,
+        #[clap(long, help = "Minimum reward tokens per publisher")]
```
Comment (guibescos) [inline staking/cli/src/cli.rs:116]: add a comment saying this is with full decimals?

File: staking/cli/src/instructions.rs [hunk 5]
```diff
@@ -870,6 +878,191 @@ pub fn update_y(rpc_client: &RpcClient, signer: &dyn Signer, y: u64) {
...
+        .collect::<Vec<_>>();
+
+    let current_epoch = get_current_epoch(rpc_client);
+
+    let data: Vec<DynamicPositionArray> = data
+        .iter_mut()
+        .filter_map(|positions| {
+            let acc = positions.to_dynamic_position_array();
+            let valid = acc
+                .get_target_exposure(&Target::IntegrityPool, current_epoch)
```
Comment (guibescos) [inline staking/cli/src/instructions.rs:1048]: Not sure about using get_target_exposure here. get_target_exposure ignores UNLOCKED positions but these can have unclaimed rewards

File: staking/cli/src/instructions.rs (hunk 1 above)
Comment (keyvankhademi) [reply staking/cli/src/instructions.rs:880]: N/A anymore

File: staking/cli/src/instructions.rs (hunk 3 above)
Comment (keyvankhademi) [reply staking/cli/src/instructions.rs]: N/A anymore

**Pull request merged**